### PR TITLE
Rerequest reviewers on pr closing

### PR DESCRIPTION
**Pull request by PaliC**
Implements the first half of the strategy discussed in https://github.com/pytorch/pytorch/issues/113453 where we rerequest a review when a contributor without write permissions has a pr get reverted.

For the second half of the strategy, I believe it has to be done on the test-infra side with pytorchbot. Though correct me if I'm wrong.

Testing:
Made sure the following worked on 

[github.com/pytorch/pytorch/pulls/113719](https://github.com/pytorch/pytorch/pull/113719)

```

if __name__ == "__main__":
    gh_dismiss_pr_reviews("pytorch", "pytorch", 113719, "test [num]")
    gh_request_pr_reviewers("pytorch", "pytorch", 113719, ["PaliC"])
```

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at fc98a58</samp>

Update the `trymerge.py` script and the `github_utils.py` module to improve the automation of pull request testing and merging. Add new functions to check write access, dismiss reviews, and request reviewers for pull requests.

### CODE REVIEW

**Comment by kit1980** [comment]
Does it do what's expected, require new approval?
I just tried to request my review on a PR I already approved, the old approval stays.
You probably what to dismiss review instead https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#dismiss-a-review-for-a-pull-request instead if requesting

File: .github/scripts/github_utils.py
Comment (ZainRizvi) [inline .github/scripts/github_utils.py:184]: Can you improve the type hints to better explain what this return value contains?

And actually, is this method even used?

File: .github/scripts/trymerge.py
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:1811]: Let's move this into the if block below to avoid an unnecessary read request to Github when the author does have write permissions

File: .github/scripts/trymerge.py
Comment (ZainRizvi) [inline .github/scripts/trymerge.py:1813]: nit: You might be able to update the gql for GH_COMMIT_AUTHORS_FRAGMENT to return the author's write access as well, removing the need for this api call to github too

For context, in the past we've had issues with being throttled by github for making too many api calls, so it's a good practice to be mindful about when we have to make them

In this case, it's okay to not implement the above since this is only invoked during reverts (which is a relatively rare occurrence), but we'd want to be careful about adding logic like this to the normal merge operation

**Review by ZainRizvi — APPROVED** [review]

**Comment by ZainRizvi** [comment]
> Does it do what's expected, require new approval? I just tried to request my review on a PR I already approved, and the old approval stays. You probably what to dismiss review instead https://docs.github.com/en/rest/pulls/reviews?apiVersion=2022-11-28#dismiss-a-review-for-a-pull-request

Please do validate whether the dismissal actually works!

File: .github/scripts/github_utils.py
Comment (PaliC) [reply .github/scripts/github_utils.py:184]: I think it was an artifact of git messing up a merge.

**Comment by kit1980** [comment]
Have you verified it works?
On https://github.com/malfet/deleteme, for example?

**Review by malfet — CHANGES_REQUESTED** [review]
This feels like a wrong integration point to be frank.
IMO it should be done at the PytorchBot rather than at mergebot level
Even without revert, if PR was closed and then re-opened we should strip reviews from it

**Comment by kit1980** [comment]
> Even without revert, if PR was closed and then re-opened we should strip reviews from it

Hm, why?

**Pull request closed without merging**
